### PR TITLE
Fix GLTFLoader: cloned material sharing the same color object

### PR DESCRIPTION
**Pull request by but0n**

### CODE REVIEW

**Comment by but0n** [comment]
And also emissive and specular, those params are copy by reference.

File: examples/js/loaders/GLTFLoader.js [hunk 1]
```diff
@@ -845,6 +845,12 @@ THREE.GLTFLoader = ( function () {
 
 				}
 
+				if ( source.color != undefined ) {
```
Comment (donmccurdy) [inline examples/js/loaders/GLTFLoader.js]: nit: Let's either use `!==` here, or just `if ( source.color )`.

File: examples/js/loaders/GLTFLoader.js [hunk 2]
```diff
@@ -845,6 +845,24 @@ THREE.GLTFLoader = ( function () {
...
+					target.specular = source.specular.clone();
+
+				}
+
+				if ( source.emissive ) {
+
+					target.emissive = source.emissive.clone();
+
+				}
+
```
Comment (takahirox) [inline examples/js/loaders/GLTFLoader.js]: No strong opinion but how about this? Less LOC addition.

```javascript
for ( var i = 0, il = params.length; i < il; i ++ ) {

    var value = source[ params[ i ] ];

  	target[ params[ i ] ] = value.clone ? value.clone() : value;

}
```

Comment (but0n) [reply examples/js/loaders/GLTFLoader.js]: What if the target value is a texture?

Comment (takahirox) [reply examples/js/loaders/GLTFLoader.js]: Oops, I was missing texture, thanks. So it'll be like this?

```javascript
for ( var i = 0, il = params.length; i < il; i ++ ) {

    var value = source[ params[ i ] ];
    target[ params[ i ] ] = value.isColor ? value.clone() : value;

}
```

Comment (but0n) [reply examples/js/loaders/GLTFLoader.js]: Looks perfect!

Comment (takahirox) [reply examples/js/loaders/GLTFLoader.js]: You seem being missing this line

```javascript
var value = source[ params[ i ] ];
```

Comment (but0n) [reply examples/js/loaders/GLTFLoader.js]: Oops, my mistake

**Review by takahirox — APPROVED** [review]

**Comment by mrdoob** [comment]
Thanks!

File: examples/js/loaders/GLTFLoader.js (hunk 2 above)
Comment (but0n) [reply examples/js/loaders/GLTFLoader.js]:
```js
      target[  params[ i ]  ] = ( value && value.isColor ) ? value.clone() : value;
```
I think it should like this, if current texture is null

Comment (takahirox) [reply examples/js/loaders/GLTFLoader.js]: Good catch! Would you like to make PR fixing that?

Comment (but0n) [reply examples/js/loaders/GLTFLoader.js]: Ok, I will make a PR.

**Pull request merged**